docs/architecture/Data_caching_lineage.md

```diff
@@ -0,0 +1,297 @@
+# Machine Learning Data Caching and Lineage
```

Comment:
Your PR mentions CV, but this document doesn't. In "Recommended Data Caching Strategy" you mention CV but don't talk about other domains. I would either make it clear that this guide is predominantly focused on CV (but may have applications elsewhere) or say that this guide is general but some of the recommendations will be specific to CV since that's the domain in which it's been tested.